pkg/spdk/replica.go

```diff
@@ -377,16 +387,33 @@ func constructSnapshotMap(replicaName string, rootSvcLvol *Lvol, bdevLvolMap map
 }
 
 func (r *Replica) Create(spdkClient *spdkclient.Client, exposeRequired bool, superiorPortAllocator *util.Bitmap) (ret *spdkrpc.Replica, err error) {
+	updateRequired := true
+
 	r.Lock()
-	defer r.Unlock()
+	defer func() {
+		r.Unlock()
+
+		if updateRequired {
+			r.UpdateCh <- nil
+		}
+	}()
 
-	if r.State != ReplicaStatePending && r.State != ReplicaStateStopped {
-		return nil, fmt.Errorf("invalid state %s for replica %s creation", r.Name, r.State)
+	if r.State == types.InstanceStateRunning {
+		updateRequired = false
+		return nil, grpcstatus.Errorf(grpccodes.AlreadyExists, "replica %v already exists and running", r.Name)
+	}
+	if r.State == types.InstanceStateStopped && !exposeRequired {
+		updateRequired = false
+		return nil, grpcstatus.Errorf(grpccodes.AlreadyExists, "replica %v already exists without expose", r.Name)
+	}
```

Comment:
No need to error out here.
It is for the local lvol case. Still need to return replica, set it to running and update.

Reply:
Here I would like to make the behavior consistent:
1. If the replica is running/exposed, trying to expose it again would error out.
2. If the local replica is created but does not need to be exposed, trying to create it without expose again would error out as well.

Reply:
For the a running lvol, the previous check `if r.State == types.InstanceStateRunning { .. }` can error out.
 
But for the lvol, we still transient the state from `stopped `to `running`.

Reply:
After discussing withe Derek, we will remove this check. 
`InstanceStateStopped` means the replica releases ports/hasn't got ports. And it's valid for API `Create`